### PR TITLE
unistd/stdio: `resolve_path()`, `fopen()` and `open()` errno fixes

### PR DESCRIPTION
**Pull request by niewim19**
## Description

 - corrections in errno returned from `resolve_path()` with compliance to posix `resolve()` function
 - ~~corrections in errno returned from `open()` with compliance to posix~~ **DROPPED**
 - corrections in errno returning from `fopen()` with compliance to posix

## Motivation and Context

 - Changes in `resolve_path()` for accordance with documentation:
> The realpath() function shall fail if:
    (...)
    [EINVAL] The file_name argument is a null pointer.
    (...)
    [ENOENT] A component of file_name does not name an existing file or file_name points to an empty string.

 - ~~Add handling of `path = ""` in `open()` for accordance with:~~ **DROPPED**
> ~~open() shall fall if:
    [ENOENT] O_CREAT is not set and a component of path does not name an existing file, or O_CREAT is set and a component of the path prefix of path does not name an existing file, **or path points to an empty string**.~~

 - Removed `path == NULL` check in `fopen()` (it is checked in `resolve_path()` later on) for accordance with:
> The fopen() function may fail if:
    [EINVAL] The value of the mode argument is not valid. 
    (...)
    [ENOENT] The mode string begins with 'r' and a component of pathname does not name an existing file, or mode begins with 'w' or 'a' and a component of the path prefix of pathname does not name an existing file, or pathname is an empty string. 
    

Related issues:
 - https://github.com/phoenix-rtos/phoenix-rtos-project/issues/256
 - https://github.com/phoenix-rtos/phoenix-rtos-project/issues/287
 - https://github.com/phoenix-rtos/phoenix-rtos-project/issues/288
 - https://github.com/phoenix-rtos/phoenix-rtos-project/issues/296

**Note:** not all issues mentioned above will be fully solved by this change as some issues are collections of multiple ones found.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: ia-32 generic.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: unistd/file.c [hunk 1]
```diff
@@ -120,9 +120,9 @@ int open(const char *filename, int oflag, ...)
 	mode = va_arg(ap, mode_t);
 	va_end(ap);
 
-	if (oflag & (O_WRONLY | O_RDWR)) {
+	if (oflag & (O_WRONLY | O_RDWR | O_RDONLY)) {
 		if ((err = stat(filename, &st)) < 0) {
-			if (errno != ENOENT)
+			if (errno != ENOENT || (oflag & O_RDONLY))
```
Comment (agkaminski) [inline unistd/file.c:125]: Why this change? I don't see a correlation with PR message:
"
    Add handling of path = "" in open() for accordance with:

    open() shall fall if:
    [ENOENT] O_CREAT is not set and a component of path does not name an existing file, or O_CREAT is set and a component of the path prefix of path does not name an existing file, or path points to an empty string.
"

File: unistd/dir.c [hunk 2]
```diff
@@ -271,7 +271,7 @@ char *resolve_path(const char *path, char *resolved_path, int resolve_last_symli
 	size_t pathlen;
 
 	if (!path || !path[0]) {
-		errno = EINVAL;
+		errno = (!path) ? EINVAL : ENOENT;
```
[the remark this answers is not in the document]
Comment (niewim19) [reply unistd/dir.c]: Changed to two `if`s, thanks!

File: unistd/file.c (hunk 1 above)
Comment (niewim19) [reply unistd/file.c]: Yes, this was valid without/before `resolve_path()` changes, sorry for that. 

Change in _unistd/file.c_ dropped.

**Review by agkaminski — APPROVED** [review]
